discretenet/problems/school_bus_scheduling/school_bus_scheduling.py

```diff
@@ -0,0 +1,185 @@
+
+        model.c1 = pyo.Constraint(model.Rs, rule=c1_rule)
+
+        def c2_rule(m, s):
+            return pyo.quicksum(m.y[t, s] for t in m.T) == 1
+
+        model.c2 = pyo.Constraint(model.S, rule=c2_rule)
+
+        def c3_rule(m, t, s, r):
+            return pyo.quicksum(
+                m.x[t_prime, s, r] for t_prime in range(1, t + 1)
+            ) <= pyo.quicksum(
+                m.y[t_prime, s] for t_prime in range(1, min(t + time_window + 1, 120))
+            )
+
+        model.c3 = pyo.Constraint(model.T, model.Rs, rule=c3_rule)
+
+        def c4_rule(m, t, s, r):
+            return pyo.quicksum(
+                m.y[t_prime, s] for t_prime in range(1, t + 1)
+            ) <= pyo.quicksum(m.x[t_prime, s, r] for t_prime in range(1, t + 1))
+
+        model.c4 = pyo.Constraint(model.T, model.Rs, rule=c4_rule)
+
+        def c5_rule(m, t):
+            return (
+                pyo.quicksum(
+                    pyo.quicksum(
+                        m.x[t_prime, s, r]
+                        for t_prime in range(t, min(t + m.R[s][r], all_time[-1] + 1))
+                    )
+                    for s, r in m.Rs
+                )
+                <= model.z
+            )
+
+        model.c5 = pyo.Constraint(model.T, rule=c5_rule)
+        self.model = model
+
+    def get_name(self):
+        return self.name
+
+    def get_parameters(self):
+        # to be implemented
+        return None
+
+
+class SchoolBusSchedulingGenerator(Generator[SchoolBusSchedulingProblem]):
+    def __init__(
+        self,
+        random_seed: int = 42,
+        path_prefix: Union[str, Path] = None,
+        num_routes=6,
+        max_time=120,
+        # school_start_interval=5,
+        num_schools=5,
+        time_window=20,
+        route_length_avg=30,
+        route_length_std=10,
+    ):
+        """
+        Initialize the school bus scheduling problem generator instance
+        following https://arxiv.org/abs/1803.09040v2
+        :param random_seed: The random seed to use
+        :param path_prefix: Path prefix to pass to instance ``save()`` methods
+            during batch generation. Must be set as a public instance attribute,
+            to be changed by the user after generator instantiation if desired.
+        :param num_routes: number of bus routes per school
+        :param max_time: total number of time slots ex: 120 for 2 hours
+        :param school_start_interval: school start time intervals
+            ex: 10 for 10, 20, 30 ... (Not implemented)
+        :param num_schools: number of schools
+        :param time_window: time window of acceptable bus arrival time
+        :param route_length_avg: average route length (time)
+        :param route_length_std: route length standard deviation
+        """
+        super().__init__(random_seed, path_prefix)
+        self.num_routes = num_routes
+        self.all_time = range(1, max_time)
+        # self.school_start_times = range(
```

Comment:
Is this no longer necessary (along with the corresponding argument and docstring line)?

Reply:
This is a feature that I wanted to implement but didn't have the time to, essentially it further restricts the school start time. I have removed them for now since they are not a mandatory part of the problem formulation.